pkg/storage/stores/local/downloads.go

```diff
@@ -212,18 +209,18 @@ func (s *Shipper) downloadFilesForPeriod(ctx context.Context, period string, fc
 
 		totalFilesSize += stat.Size()
 
```

Comment:
I think the lock can be held here rather than implicitly require the caller to do so.

Reply:
I am taking the lock from the caller because this function is called in a goroutine. The goroutine could be waiting to be scheduled and some other goroutine could take the lock. While I am checking the `ready` channel in reads but it would be error-prone to rely only on that because someone else might forget to check the channel. The contract is simple now, all the public methods take care of locking.